### PR TITLE
Don't copy primary variables into element context

### PR DESCRIPTION
**Pull request by akva2**
This seems to yield a consistent speedup of assembly in my tests.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.998    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.009    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.012    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.004    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1.007    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.009    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1576

File: opm/models/discretization/common/fvbaseelementcontext.hh [hunk 1]
```diff
@@ -502,7 +494,7 @@ public:
      */
     void restoreIntensiveQuantities(unsigned dofIdx)
     {
-        dofVars_[dofIdx].priVars[/*timeIdx=*/0] = priVarsStashed_;
+        dofVars_[dofIdx].priVars[/*timeIdx=*/0] = &priVarsStashed_;
```
Comment (blattms) [inline opm/models/discretization/common/fvbaseelementcontext.hh:497]: This looks a bit dangerous. Are we sure that the entries in `priVarsStashed_` always live longer than the pointers to it that we stored here?

**Review by blattms — COMMENTED** [review]
Not for the faint of heart. Hope there are no lifetime issues that may cause dangling pointers.

File: opm/models/discretization/common/fvbaseelementcontext.hh (hunk 1 above)
Comment (akva2) [reply opm/models/discretization/common/fvbaseelementcontext.hh:497]: We don't store pointers there. the stash is a class member we copy into on stash call. Thus both stash and dofvars are class member so they should have similar lifetimes.

Comment (blattms) [reply opm/models/discretization/common/fvbaseelementcontext.hh:497]: Thanks. I see.

**Pull request merged**